### PR TITLE
Fix a bug with provisioning_uri 

### PR DESCRIPTION
**Pull request by ResultsMayVary**
If there is no email it would crash due to nil.

### CODE REVIEW

**Comment by robertomiranda** [comment]
@ResultsMayVary could you take a look on travis? the tests are failing. Also could you reproduce the bug writing a test case?

**Comment by ResultsMayVary** [comment]
Sorry for the delay. These changes should remedy all your issues. Test cases have been added and I improved some checks in a couple spots. Let me know!

File: lib/active_model/one_time_password.rb [hunk 1]
```diff
@@ -46,7 +46,7 @@ def authenticate_otp(code, options = {})
           result = hotp.verify(code, otp_counter)
           if result && options[:auto_increment]
             self.otp_counter += 1
-            save if !new_record?
+            save if respond_to?(:new_record) && !new_record?
```
Comment (robertomiranda) [inline lib/active_model/one_time_password.rb]: I was wondering, why is needed to call respond_to method here?

Comment (ResultsMayVary) [inline lib/active_model/one_time_password.rb]: Test cases don't respond to :new_record, so I figured this was the safest fix.

**Pull request merged**